### PR TITLE
Pim 5897

### PR DESCRIPTION
**Pull request by juliensnz**
**Description (for Contributor and Core Developer)**

Fix the does not contains filter which was not filtering on products with empty values

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Y
| Added Behats                      | Y
| Changelog updated                 | N
| Review and 2 GTM                  | N
| Micro Demo to the PO (Story only) | NA
| Migration script                  | NA
| Tech Doc                          | NA

### CODE REVIEW

**Comment by BitOne** [comment]
GTM

File: src/Pim/Bundle/CatalogBundle/Doctrine/ORM/Filter/StringFilter.php [hunk 1]
```diff
@@ -85,16 +85,33 @@ public function addAttributeFilter(
...
                     $this->qb->expr()->neq($backendField, $this->qb->expr()->literal(''))
                 );
+                $this->qb->innerJoin(
+                    $this->qb->getRootAlias() . '.values',
+                    $joinAlias,
+                    'WITH',
+                    $condition
+                );
+            } elseif (Operators::DOES_NOT_CONTAIN === $operator) {
+                $whereCondition = $this->prepareCondition($backendField, $operator, $value) .
```
Comment (pierallard) [inline src/Pim/Bundle/CatalogBundle/Doctrine/ORM/Filter/StringFilter.php]: Should you add parenthesis around this condition ? (Not completely sure how it works, sorry dumb question)

**Comment by pierallard** [comment]
Please add a test or test by yourself than you can use 2 conditions (1 DOES NOT CONTAIN and 1 "other") then GTM

File: src/Pim/Bundle/CatalogBundle/Doctrine/ORM/Filter/StringFilter.php (hunk 1 above)
Comment (juliensnz) [inline src/Pim/Bundle/CatalogBundle/Doctrine/ORM/Filter/StringFilter.php]: the addWhere does it for us :)

**Comment by juliensnz** [comment]
@pierallard this is done in the modified behat :)

**Pull request merged**